src/neuroglancer/ui/segment_select_tools.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * /**
+ * @license
+ * This work is a derivative of the Google Neuroglancer project,
+ * Copyright 2016 Google Inc.
+ * The Derivative Work is covered by
+ * Copyright 2021 Howard Hughes Medical Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {SegmentationUserLayer} from 'neuroglancer/segmentation_user_layer';
+import {removeChildren} from 'neuroglancer/util/dom';
+import {makeToolActivationStatusMessageWithHeader, registerLayerTool, Tool, ToolActivation} from 'neuroglancer/ui/tool';
+import { EventActionMap } from '../util/event_action_map';
+
+export const SELECT_SEGMENTS_TOOLS_ID = 'selectSegments';
+
+const selectEvent = 'mousedown0';
+const deselectEvent = 'mousedown2';
+const SELECT_SEGMENTS_INPUT_EVENT_MAP = EventActionMap.fromObject({
+  [`at:shift?+${selectEvent}`]: 'drag-select-segments',
+  [`at:shift?+${deselectEvent}`]: 'drag-deselect-segments',
+  'at:shift?+mouseup0': 'deactivate-drag-select-segments',
+  'at:shift?+mouseup2': 'deactivate-drag-deselect-segments',
+});
+
+enum ToolState {
+  IGNORE,
+  SELECT,
+  DESELECT,
+};
+
+export class SelectSegmentsTool extends Tool<SegmentationUserLayer> {
+  constructor(layer: SegmentationUserLayer, private state: ToolState) {
```

Comment:
It might make more sense for `state` to be a local variable within `activate` rather than a member of `SelectSegmentsTool`.